### PR TITLE
StyleKit

### PR DESCRIPTION
**Pull request by TadeasKriz**
Add a new styling library with a simple DSL, low memory footprint and great performance.

### CODE REVIEW

File: SwiftKit/Router/Methods.swift [hunk 1]
```diff
@@ -198,7 +199,7 @@ public class PATCH<IN, OUT>: BaseEndpoint<IN, OUT> {
     :param: IN The input type
     :param: OUT The output type
 */
-public class TRACE<IN, OUT>: BaseEndpoint<IN, OUT> {
+public final class TRACE<IN, OUT>: BaseEndpoint<IN, OUT> {
```
Comment (matoushybl) [inline SwiftKit/Router/Methods.swift:202]: This shouldn't be in this PR in my opinion.

Comment (TadeasKriz) [inline SwiftKit/Router/Methods.swift:202]: Damn, probably a rebase issue?

File: SwiftKit/StyleKit/StyleDeclarationPart.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    func collapse<T: Styleable, S: SequenceType
+        where S.Generator.Element == String>(type: T.Type, names: S, styling: T -> ())
+    {
+        let target = StyleTarget(type: type, names: Set(names))
+        let typeErasedStyling = Style.typeEraseStylingFunction(styling)
+        let blueprint = StyleBlueprint(firstTarget: target, styling: typeErasedStyling)
+        
+        collapse(blueprint)
+    }
+}
```
Comment (matoushybl) [inline SwiftKit/StyleKit/StyleDeclarationPart.swift]: Github shows warning about newlines on file ends.

Comment (TadeasKriz) [inline SwiftKit/StyleKit/StyleDeclarationPart.swift]: I think that's not an issue, but an information, isn't it?

Comment (matoushybl) [inline SwiftKit/StyleKit/StyleDeclarationPart.swift]: Shouldn't matter, only Diff doesn't like it.

**Pull request merged**